src/SFA.DAS.Commitments.Api.Client/HttpCommitmentHelper.cs

```diff
@@ -178,5 +178,11 @@ public Task<string> GetBulkuploadFile(string url)
 
             return JsonConvert.DeserializeObject<TransferRequest>(content);
         }
+
+        public async Task<T> GetUrlResult<T>(string url)
```

Comment:
It would be nice to switch the existing methods in this class over to this new helper method.

Reply:
Completely agree, I did a load of work to remove the IHttpCommitmentHelper altogether, but this meant we had to refactor the HttpClient class (create a new version, but this is library code).

This is a separate branch. The choice here is we improve the Helper function or better still just get rid of it